lib/Sema/CSDiag.cpp

```diff
@@ -3345,6 +3352,33 @@ class ArgumentMatcher : public MatchCallArgumentListener {
     Diagnosed = true;
   }
 
+  bool isPropertyWrapperImplicitInit() {
+    auto TE = dyn_cast<TypeExpr>(FnExpr);
+    if (!TE)
+      return false;
+
+    auto instanceTy = TE->getInstanceType();
+    if (!instanceTy)
+      return false;
+
+    auto nominalDecl = instanceTy->getAnyNominal();
+    if (!nominalDecl)
+      return false;
+
+    if (!nominalDecl->getAttrs().hasAttribute<PropertyWrapperAttr>())
+      return false;
+
+    auto parent = CandidateInfo.CS.getParentExpr(FnExpr);
+    if (!parent)
+      return false;
+
+    auto CE = dyn_cast<CallExpr>(parent);
+    if (!CE)
+      return false;
+
+    return CE->isImplicit();
```

Comment:
This is a little overboard I think :) How about this:

```c++
auto NTD = instanceTy->getAnyNominal();
if (!(NTD && NTD->getAttrs().hasAttribute<PropertyWrapperAttr>()))
      return false;

if (auto *parentExpr = CandidateInfo.CS.getParentExpr(FnExpr)) {
   return parentExpr->isImplicit() && isa<CallExpr>(parentExpr);
}

return false;
```

Reply:
Ok done!